connector/extensions/web-service/src/main/java/org/oaebudt/edc/web/service/MetadataValidator.java

```diff
@@ -0,0 +1,68 @@
+package org.oaebudt.edc.web.service;
+
+import java.net.URI;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+public class MetadataValidator {
+
+    private static final Pattern EMAIL_REGEX = Pattern.compile(
+            "^[A-Z0-9._%+-]+@[A-Z0-9.-]+\\.[A-Z]{2,}$", Pattern.CASE_INSENSITIVE
+    );
+
+    public static ValidationResult validate(Map<String, Object> metadata) {
+        ValidationResult result = ValidationResult.ok();
+
+        result.merge(validateRequiredString(metadata, "legalOrganizationName"));
+        result.merge(validateRequiredString(metadata, "countryOfOrganization"));
+        result.merge(validateRequiredString(metadata, "contactPerson"));
+        result.merge(validateRequiredString(metadata, "dataProcessingDescription"));
+        result.merge(validateRequiredString(metadata, "qualityAssuranceMeasures"));
+        result.merge(validateRequiredString(metadata, "dataLicensingTerms"));
+
+        result.merge(validateUrl(metadata, "organizationWebsite"));
+        result.merge(validateEmail(metadata, "contactEmail"));
+
+        result.merge(validateLevel(metadata, "dataAccuracyLevel"));
+        result.merge(validateLevel(metadata, "dataGenerationTransparencyLevel"));
+        result.merge(validateLevel(metadata, "dataDeliveryReliabilityLevel"));
+        result.merge(validateLevel(metadata, "dataFrequencyLevel"));
+        result.merge(validateLevel(metadata, "dataGranularityLevel"));
+        result.merge(validateLevel(metadata, "dataConsistencyLevel"));
+
+        return result;
```

Comment:
nit, the validations could be a stream on which you could apply a `reduce` function to merge them.

Reply:
Done!!